apps/hash-api/src/storage/index.ts

```diff
@@ -138,7 +138,7 @@ const getFileEntity = async (
         ],
       },
       graphResolveDepths: zeroedGraphResolveDepths,
-      temporalAxes: fullDecisionTimeAxis,
+      temporalAxes: currentTimeInstantTemporalAxes,
```

Comment:
@CiaranMn this was throwing an error when viewing file entities, as we are now updating them so expect there to be more than one version in the history. Not sure why we were using `fullDecisionTimeAxis` here in the first place.

Reply:
If the file entity is updated by replacing the file, it gets a new key identifying it in storage – the `key` is passed as part of the query, and includes the edition id (a timestamp).

We need `fullDecisionTimeAxis` to be able to retrieve earlier versions of files. The error is happening now because we are changing the file entity without changing the key, which were doing previously.

This can be tested by going to the file entity's page, saving the key, and clicking the button to replace the image. I assume that attempting to visit the URL for the previous edition's key will now crash.

Probably the easiest answer is to remove the check for unexpected entities and take whatever the latest is with the specified key.

Reply:
Ah thanks for explaining, I've addressed this in https://github.com/hashintel/hash/pull/3929/commits/20f0f51ef474d80409067407a785c90d9be30695